### PR TITLE
[Docs/Service Level Readme] Fix DirectoryPath logic

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/common/scripts/Service-Level-Readme-Automation.ps1 [hunk 1]
```diff
@@ -73,22 +73,38 @@ foreach($moniker in $monikers) {
...
-          }
-        }
+
+      if (!$onboardedPackages.ContainsKey($pkgKey)) {
+        continue
+      }
+
+      $package = $onboardedPackages[$pkgKey]
+
+      if (!$package) {
```
Comment (weshaggard) [inline eng/common/scripts/Service-Level-Readme-Automation.ps1:87]: Given the check to make sure the key exists do you ever expect the value to be null?

File: eng/common/scripts/Verify-RequiredDocsJsonMembers.ps1 [hunk 2]
```diff
@@ -91,6 +91,13 @@ function Test-RequiredDocsJsonMembers($moniker) {
       $script:FoundError = $true
     }
 
+    if ($fileObject.PSObject.Members.Name -contains 'DirectoryPath') {
```
Comment (weshaggard) [inline eng/common/scripts/Verify-RequiredDocsJsonMembers.ps1:94]: Should we treat the absence of the DirectoryPath property as a failure as well?

**Review by weshaggard — COMMENTED** [review]
Couple comments otherwise looks good.

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/Service-Level-Readme-Automation.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/common/scripts/Service-Level-Readme-Automation.ps1:87]: Yes. The value can be null in cases where a package is onboarded in the ci configuration for the docs but that package has no corresponding metadata JSON file. 

We should clean those up, too. https://github.com/Azure/azure-sdk-tools/issues/6283

File: eng/common/scripts/Verify-RequiredDocsJsonMembers.ps1 (hunk 2 above)
Comment (danieljurek) [reply eng/common/scripts/Verify-RequiredDocsJsonMembers.ps1:94]: This is done

**Comment by danieljurek** [comment]
This should've merged a while back. I'll merge it soon but not during release week.

**Pull request merged**